coprocessor/fhevm-engine/host-listener/src/database/ingest.rs

```diff
@@ -0,0 +1,103 @@
+use std::collections::HashSet;
+
+use alloy::primitives::Address;
+use alloy::rpc::types::Log;
+use alloy::sol_types::SolEventInterface;
+use fhevm_engine_common::types::Handle;
+use tracing::{error, info};
+
+use crate::cmd::block_history::BlockSummary;
+use crate::contracts::{AclContract, TfheContract};
+use crate::database::tfhe_event_propagate::{
+    acl_result_handles, tfhe_result_handle, Database, LogTfhe,
+};
+
+pub struct BlockLogs<T> {
+    pub logs: Vec<T>,
+    pub summary: BlockSummary,
+    pub catchup: bool,
+}
+
+pub async fn ingest_block_logs(
+    chain_id: u64,
+    db: &mut Database,
+    block_logs: &BlockLogs<Log>,
+    acl_contract_address: &Option<Address>,
+    tfhe_contract_address: &Option<Address>,
+) -> Result<(), sqlx::Error> {
+    let mut tx = db.new_transaction().await?;
+    let mut is_allowed = HashSet::<Handle>::new();
+    let mut tfhe_event_log = vec![];
+    let block_hash = block_logs.summary.hash;
+    let block_number = block_logs.summary.number;
+
+    for log in &block_logs.logs {
+        let current_address = Some(log.inner.address);
+        let is_acl_address = &current_address == acl_contract_address;
+        if acl_contract_address.is_none() || is_acl_address {
+            if let Ok(event) =
+                AclContract::AclContractEvents::decode_log(&log.inner)
+            {
+                info!(acl_event = ?event, "ACL event");
+                let handles = acl_result_handles(&event);
+                for handle in handles {
+                    is_allowed.insert(handle.to_vec());
+                }
+                db.handle_acl_event(
+                    &mut tx,
+                    &event,
+                    &log.transaction_hash,
+                    chain_id,
+                    block_hash.as_ref(),
+                    block_number,
+                )
+                .await?;
```

Comment:
Just a question, what happens in this case: if the db.handle_acl_event fails ? 
What is the behaviour ? Is the insertion for all thoses logs are retried ?